querydsl-collections/src/main/java/com/querydsl/collections/DefaultQueryEngine.java

```diff
@@ -188,7 +188,7 @@ private List evaluateSingleSource(QueryMetadata metadata, Map<Expression<?>,
                 order(metadata, sources, list);
             }
             // projection
-            if (metadata.getProjection().size() > 1 || !metadata.getProjection().get(0).equals(source)) {
+            if (!metadata.getProjection().equals(source)) {
```

Comment:
If you move the annotation to the interface, this would highlight as a possible null pointer dereference.